test/writer.js

```diff
@@ -41,39 +41,51 @@ describe('#unit Check database restore writer', function() {
     nock.cleanAll();
   });
 
-  it('should complete successfully', function(done) {
+  it('should complete successfully', async function() {
     nock(dbUrl)
       .post('/_bulk_docs')
       .reply(200, []); // success
 
-    testLinestream()
-      .pipe(writer(db, 500, 1, noopEmitter))
-      .on('error', function(err) {
-        done(err);
-      })
-      .on('finished', function(data) {
-        assert.strictEqual(data.total, 15);
-        assert.ok(nock.isDone());
-        done();
-      });
+    return new Promise((resolve, reject) => {
+      testLinestream()
+        .pipe(writer(db, 500, 1, noopEmitter))
+        .on('error', function(err) {
+          reject(err);
+        })
+        .on('finished', function(data) {
+          try {
+            assert.strictEqual(data.total, 15);
+            assert.ok(nock.isDone());
+            resolve();
+          } catch (err) {
+            reject(err);
+          }
+        });
+    });
   });
 
-  it('should terminate on a fatal error', function(done) {
+  it('should terminate on a fatal error', async function() {
     nock(dbUrl)
       .post('/_bulk_docs')
       .reply(401, { error: 'Unauthorized' }); // fatal error
 
-    testLinestream()
-      .pipe(writer(db, 500, 1, noopEmitter))
-      .on('error', function(err) {
-        assert.strictEqual(err.name, 'Unauthorized');
-        assert.strictEqual(err.message, 'Access is denied due to invalid credentials.');
-        assert.ok(nock.isDone());
-        done();
-      });
+    return new Promise((resolve, reject) => {
```

Comment:
In the rejection case I think a pipeline like this could work:
```
    const w = writer(db, 500, 1, noopEmitter);
    return assert.rejects(
      pipeline(fs.createReadStream('./test/fixtures/animaldb_expected.json'), liner(), w),
      (err) => {
        assert.strictEqual(err.name, 'Unauthorized');
        assert.strictEqual(err.message, 'Access is denied due to invalid credentials.');
        assert.ok(nock.isDone());
        return true;
      });
```

ofc the pipeline is the `const { pipeline } = require('node:streams/promises');` form.